src/components/counter/counter.test.tsx

```diff
@@ -0,0 +1,29 @@
+import { describe, expect, test } from 'vitest'
+import { render } from '@testing-library/react'
+import '@testing-library/jest-dom'
+import { Counter } from './counter'
+import userEvent from '@testing-library/user-event'
+
+const user = userEvent.setup()
+
+describe('Counter Component', () => {
+  test('Test that component renders initial value provided in props', () => {
+    const { getByText } = render(<Counter initialValue={7} />)
+    const renderedValue = getByText('7')
+    expect(renderedValue).toBeInTheDocument()
+  })
+  test('Test that a click event on "decrement" button decrements the displayed value', async () => {
+    const { getByText } = render(<Counter initialValue={7} />)
+    const decrementButton = getByText('Decrement')
+    await user.click(decrementButton)
+    const decrementedValue = getByText('6')
+    expect(decrementedValue).toBeInTheDocument()
+  })
```

Comment:
`userEvent` should be added here:

```suggestion
  test('Test that a click event on "decrement" button decrements the displayed value', async () => {
    const user = userEvent.setup()
    const { getByText } = render(<Counter initialValue={7} />)
    const decrementButton = getByText('Decrement')
    await user.click(decrementButton)
    const decrementedValue = getByText('6')
    expect(decrementedValue).toBeInTheDocument()
  })
```